nexus/tests/integration_tests/endpoints.rs

```diff
@@ -622,496 +714,751 @@ lazy_static! {
+                        serde_json::to_value(&*DEMO_GLOBAL_IMAGE_CREATE).unwrap()
+                    ),
+                    visibility: Visibility::PublicPermissionRequired,
+                },
             ],
         },
 
         VerifyEndpoint {
             url: &*DEMO_GLOBAL_IMAGE_URL,
-            visibility: Visibility::Protected,
             allowed_methods: vec![
-                AllowedMethod::Get,
-                AllowedMethod::Delete,
+                AllowedMethod {
+                    method: MethodAndBody::Get,
+                    visibility: Visibility::PublicNoPermissionRequired,
+                },
+                AllowedMethod {
+                    method: MethodAndBody::Delete,
+                    visibility: Visibility::Protected, // XXX is this right?
+                },
             ],
         },
 
         /* Silo identity providers */
 
-        /*
         VerifyEndpoint {
-            url: &*IDENTITY_PROVIDERS_URL, // in ignore list
-            visibility: Visibility::Public,
+            url: &*IDENTITY_PROVIDERS_URL,
             allowed_methods: vec![
-                AllowedMethod::Get,
+                AllowedMethod {
+                    method: MethodAndBody::Get,
+                    visibility: Visibility::PublicNoPermissionRequired,
+                },
             ],
         },
-        */
+
         VerifyEndpoint {
             url: &*SAML_IDENTITY_PROVIDERS_URL,
-            visibility: Visibility::Public,
-            allowed_methods: vec![AllowedMethod::Post(
-                serde_json::to_value(&*SAML_IDENTITY_PROVIDER).unwrap(),
-            )],
+            allowed_methods: vec![
+                AllowedMethod {
+                    method: MethodAndBody::Post(
+                        serde_json::to_value(&*SAML_IDENTITY_PROVIDER).unwrap(),
+                    ),
+                    visibility: Visibility::PublicPermissionRequired,
+                },
+            ],
         },
         VerifyEndpoint {
             url: &*SPECIFIC_SAML_IDENTITY_PROVIDER_URL,
-            visibility: Visibility::Protected,
-            allowed_methods: vec![AllowedMethod::Get],
+            allowed_methods: vec![
+                AllowedMethod {
+                    method: MethodAndBody::Get,
+                    visibility: Visibility::Protected,
+                },
+            ],
+        },
+
+        /* Misc */
+
+        VerifyEndpoint {
+            url: "/session/me",
+            allowed_methods: vec![
+                AllowedMethod {
+                    method: MethodAndBody::Get,
+                    visibility: Visibility::PublicNoPermissionRequired,
+                },
+            ],
+        },
+
+        /* SSH keys */
+
+        VerifyEndpoint {
+            url: &*DEMO_SSHKEYS_URL,
+            allowed_methods: vec![
+                AllowedMethod {
+                    method: MethodAndBody::Get,
+                    visibility: Visibility::Protected, // XXX is this right?
```

Comment:
Yes, I believe this (and the ones below) is correct. There's an argument to be made that SSH public keys should be truly public (as they are on GitHub, for example), but we decided not to go that route for now. In particular, there's no was currently for one user to even address another user's keys, since we go through `/session/me`.

Reply:
What I mean is it was a little surprising that `GET /session/me/sshkeys` is expected to return a 403 - there is no endpoint to address other user's keys, this is for the current user, and I would have expected 200 OK with a body of `[]` or something like that instead.

Reply:
Oh, I misunderstood what the test was saying. You are correct: for the user that added the demo key, they should get a `200 OK` with the key that was added; for a user that has no keys, they should get `200 OK` with an empty list; and for an unauthenticated user, they should get `403 Forbidden`.